src/interactive.ts

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env node
+
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ *
+ */
+
+import * as yargs from 'yargs'
+import * as inquirer from 'inquirer'
+import * as fs from 'fs'
+import {ActionsOnGoogle, AssistResponse} from './actions-on-google'
+
+class Interactive {
+
+    async main(): Promise<void> {
+        const argv = yargs
+            .usage('Usage: action-interactive [Options] [Action] [Prompt]')
+            .example('action-interactive -c ./credentials.json "my test app"', '')
+            .option('credential', {
+                alias: 'c',
+                description: 'Your credential file path',
+                type: 'string',
+            })
+            .option('locale', {
+                alias: 'l',
+                description: 'Locale string',
+                type: 'string',
+            })
+            .version(false)
+            .argv
+        const actionName = argv._.length >= 1 ? argv._[0] : undefined
+        const prompt = argv._.length >= 2 ? argv._[1] : undefined
+        const credential = argv.credential || './credentials.json'
+        const locale = argv.locale || 'en-US'
+
+        let action: ActionsOnGoogle | undefined
+        try {
+            action = new ActionsOnGoogle(require(fs.realpathSync(credential)))
+            action.setLocale(locale)
+
+            const response = await this.start(action, actionName, prompt)
+            this.conversation(action, response)
+        } catch(e) {
+            if (action) {
+                try {
+                    await action.endConversation()
+                } catch(_) {
+                }
+            }
+            console.log(`Conversation aborted: ${e.toString()}`)
+        }
+    }
+
+    async conversation(action: ActionsOnGoogle, response: AssistResponse): Promise<void> {
+        console.log('')
+        console.log('\x1b[33m%s\x1b[0m', 'Action response:')
```

Comment:
Is this supposed to output with a different color?

Reply:
@Fleker Well, actually, I'm not sure that this approach is correct or not. At least, the current code (the color code is hard-coded) does not work on some terminals which is not supported the escape sequence, I guess. It seems that we should use some library to output something message with such escape sequence. How do you think about this?

Reply:
Colors are cool, although I'm not sure if it's worthwhile to pull in a library.

Reply:
@Fleker Ok, in this initial version, I think that we should postpone the coloring specification. I intend to remove codes to color some text. Also, I will remove the `inquirer` library from this and will apply other library which is more simple, because the `inquirer` library forcibly colors output texts. We will be able to discuss the coloring specification from a next version.

Reply:
@Fleker Ok, in this initial version, I think that we should postpone the coloring specification. I intend to remove codes to color some text. Also, I will remove the `inquirer` library from this and will apply other library which is more simple, because the `inquirer` library forcibly colors output texts. We will be able to discuss the coloring specification from a next version.